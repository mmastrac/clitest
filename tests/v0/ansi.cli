#!/usr/bin/env clitest --v0

# We can use \e reliably across platforms.

$ printf "\e[31mHello, world\e[0m\n"
! Hello, world

$ printf "\e[31mHello, world\e[0m\n"
? Hello, %{WORD}

# Not recommended, but valid
$ printf "\e[31mHello, world\e[0m\n"
? \e\[31mHello, world\e\[0m

# Ensure `set` works
set ANSI_COLORED "\e[31mHello, world\e[0m";

$ printf "$ANSI_COLORED\n"
! Hello, world

# Not recommended, but valid
$ printf "$ANSI_COLORED\n"
? \e\[31mHello, world\e\[0m
