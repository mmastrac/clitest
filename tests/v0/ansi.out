Running <root>/ansi.cli ...

printf "\e[31mHello, world\e[0m\n"
---
Hello, world
---
✅ OK

printf "\e[31mHello, world\e[0m\n"
---
Hello, world
---
✅ OK

printf "\e[31mHello, world\e[0m\n"
---
Hello, world
---
✅ OK

set ANSI_COLORED ␛[31mHello, world␛[0m

printf "$ANSI_COLORED\n"
---
Hello, world
---
✅ OK

printf "$ANSI_COLORED\n"
---
Hello, world
---
✅ OK

<root>/ansi.cli PASSED
